apps/web-mzima-client/src/app/settings/surveys/survey-item/survey-item.component.ts

```diff
@@ -344,27 +348,57 @@ export class SurveyItemComponent extends BaseComponent implements OnInit {
     }
   }
 
-  public async openConfirmModal() {
-    if (this.hasChanges()) {
+  @HostListener('window:beforeunload', ['$event']) unloadNotification($event: any): void {
+    const formHasChanged = this.hasChanges();
+    if (formHasChanged)
+      $event.returnValue = 'Form has changed! - Just to let browser popup show up';
+  }
+
+  @HostListener('window:popstate', ['$event']) onPopState() {
+    this.browserBackButtonClicked = true;
+  }
+
+  public async openConfirmModal({
+    event,
+    cancelButtonclicked,
+    navigateBack,
+  }: {
+    event: Event;
+    cancelButtonclicked?: boolean;
+    navigateBack?: boolean;
+  }) {
+    const noChangesMade = cancelButtonclicked && !this.changesMade;
+    if (noChangesMade) {
+      this.showSnackbar(this.translate.instant('notify.snackbar.no_changes_made'));
+    } else {
+      if (event) event.stopPropagation();
       const confirmed = await this.confirmModalService.open({
         title: this.translate.instant('notify.default.discard_changes'),
         description: this.translate.instant('notify.default.survey_has_not_been_saved'),
-        cancelButtonText: this.translate.instant('notify.survey.discard_changes'),
-        confirmButtonText: this.translate.instant('notify.survey.save_changes'),
-        isCancelDestructive: true,
-        isConfirmNotDestructive: true,
+        confirmButtonText: this.translate.instant('notify.survey.discard_changes'),
+        cancelButtonText: this.translate.instant('notify.survey.back_to_editing'),
+        isConfirmNotDestructive: false,
+        isCancelDestructive: false,
       });
-
-      if (confirmed) {
-        this.save();
-      } else {
+      //-- | For route canDeactivate guard | ---
+      this.modalConfirmButtonClicked = confirmed;
+      //----------------------------------------
+      if (confirmed || navigateBack) {
         this.navigateBack();
+      } else {
+        return;
       }
-    } else {
-      this.navigateBack();
     }
   }
 
+  public showSnackbar(message: string) {
+    this.snackBar.open(message, this.translate.instant('notify.snackbar.close'), {
+      duration: 3000,
+      horizontalPosition: 'center',
+      verticalPosition: 'bottom',
+    });
+  }
+
   navigateBack() {
```

Comment:
Trigger this when clicking on Cancel instead

Reply:
I will message you on slack for this.